### PR TITLE
Create room entity and related classes for room data

### PR DESCRIPTION
**Pull request by bxtsang**

### CODE REVIEW

File: webproject/src/main/java/data/Room.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  private String link;
+  private String description;
+  private int deliveryLocation; //modify to use different representation of location if needed
+  private int phoneNumber; //modify to use different representation of contact details if needed
+  private String category;
+  private String imagePath;
+  private RoomType roomType;
+  private RoomCondition roomCondition;
+
+  public Room (String id, String title, String link, String description, int deliveryLocation, int phoneNumber, String category, String imagePath, RoomType roomType, RoomCondition roomCondition) {
```
Comment (kevinluvian) [inline webproject/src/main/java/data/Room.java]: lets change this to builder class
https://dzone.com/articles/design-patterns-the-builder-pattern

**Review by kevinluvian — CHANGES_REQUESTED** [review]

File: webproject/src/main/java/data/Room.java (hunk 1 above)
Comment (bxtsang) [reply webproject/src/main/java/data/Room.java]: Changed!

**Review by MichaelaTSH — COMMENTED** [review]
Looks good!

**Review by kevinluvian — APPROVED** [review]

File: webproject/src/main/java/data/Room.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  private String link;
+  private String description;
+  private int deliveryLocation; //modify to use different representation of location if needed
+  private int phoneNumber; //modify to use different representation of contact details if needed
+  private String category;
+  private String imagePath;
+  private RoomType roomType;
+  private RoomCondition roomCondition;
+
+  private Room (Builder builder) {
```
Comment (kevinluvian) [inline webproject/src/main/java/data/Room.java]: it is better to have a function newBuilder() to create the builder, it is also useful when you want to apply default value
```
private Builder newBuilder() {
  return new Builder().setSomeParameter(defaultValue);
}
```

this way, the code will be something like
```
Room.newBuilder().setSomeParameter(...).setSomeParameter(...).build()
```

**Review by kevinluvian — CHANGES_REQUESTED** [review]

File: webproject/src/main/java/data/Room.java (hunk 2 above)
Comment (bxtsang) [reply webproject/src/main/java/data/Room.java]: Ok! should it be public to create the object in other classes?

**Review by MichaelaTSH — APPROVED** [review]

**Review by kevinluvian — APPROVED** [review]

**Pull request merged**